src/controller/python/chip/yaml/format_converter.py

```diff
@@ -20,9 +20,39 @@
 from chip.tlv import uint, float32
 import enum
 from chip.yaml.errors import ValidationError
-
-
-_HEX_PREFIX = 'hex:'
+from binascii import unhexlify
+import re
+
+
+def convert_yaml_octet_string_to_bytes(s: str) -> bytes:
+    """Convert YAML octet string body to bytes, handling any c-style hex escapes (e.g. \x5a) and hex: prefix"""
+    is_hex_string_re = r"^hex:(?P<hex_content>[A-Fa-f0-9]*)$"
+
+    hex_prefixed_match = re.match(is_hex_string_re, s)
+
+    # Step 1: handle explicit "hex:" prefix
+    if hex_prefixed_match:
+        hex_content = hex_prefixed_match.group("hex_content")
+        if (len(hex_content) % 2) != 0:
+            raise ValueError("Hex literal is not even length!")
+        return unhexlify(hex_content)
+
+    # Step 2: convert non-hex-prefixed to bytes
+    # TODO(#23669): This does not properly support utf8 octet strings. We mimic
+    # javascript codegen behavior. Behavior or javascript is:
+    #   * Octet string character >= u+0200 errors out.
+    #   * Any character greater than 0xFF has the upper bytes chopped off.
+    known_javascript_max_char_value = 0x200
+    accumulated_hex = ""
+    for char in s:
```

Comment:
I wonder if we could operate on lists:

```python
as_bytes = [ord(c) for c in s]

if any([value > 0x200 for value in as_bytes]):
    raise ValueError("Unsupport char in octet string")
    
return "".join([f"{(v & 0xFF):02x}" for v in as_bytes])
```

Reply:
I was originally going with more readible, but I guess the TODO comment above help everything that is happening here